### ISSUE TITLE
Migrate to TypeScript

### ISSUE THREAD

**Issue by vire** [realreality/reality-plugin#75]
Considering the current state of JS, there is really no reason to not use types & profit from all benefits - compile time checking, code intel, documentation, reliability, decrease in bug density etc.

@michalbcz WDYT ?

**Comment by michalbcz** [comment]
@vire yep, you are right. why not. you've got green if you want to do it :)
